code/modules/mob/living/carbon/human/species/diona.dm

```diff
@@ -86,7 +86,7 @@
 
 /datum/species/diona/handle_life(mob/living/carbon/human/H)
 	var/light_amount = 0 //how much light there is in the place, affects receiving nutrition and healing
-	var/is_vamp = H.mind?.has_antag_datum(/datum/antagonist/vampire) != null
+	var/is_vamp = !H.mind?.has_antag_datum(/datum/antagonist/vampire)
```

Comment:
```suggestion
	var/is_vamp = H.mind && H.mind.has_antag_datum(/datum/antagonist/vampire)
```
It currently checks if they are not a vamp
Also the `H.mind?.` makes it sometimes return a `null`. Which makes it more annoying since you can't just do `H.mind?.has_antag_datum(/datum/antagonist/vampire) != FALSE`

However, doing it like this will also change it into a boolean value. Like how `!!H.mind?.has_antag_datum(/datum/antagonist/vampire)` would have done.